### PR TITLE
Add support for classic builder

### PR DESCRIPTION
**Pull request by ulyssessouza**
**What I did**
Add support for classic (non buildkit) builds

/generate-artifacts

**Related issue**

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: pkg/compose/build_classic.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+			}
+			return "", cli.StatusError{Status: jerr.Message, StatusCode: jerr.Code}
+		}
+		return "", err
+	}
+
+	// Windows: show error message about modified file permissions if the
+	// daemon isn't running Windows.
+	if response.OSType != "windows" && runtime.GOOS == "windows" {
+		// if response.OSType != "windows" && runtime.GOOS == "windows" && !options.quiet {
```
[the remark this answers is not in the document]
Comment (ulyssessouza) [reply pkg/compose/build_classic.go:202]: That was testing code. Removing

File: pkg/compose/build_classic.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+		NetworkMode: options.NetworkMode,
+		ExtraHosts:  options.ExtraHosts,
+		Target:      options.Target,
+	}
+}
+
+func toMapStringStringPtr(source map[string]string) map[string]*string {
+	dest := make(map[string]*string)
+	for k, v := range source {
+		v := v
```
[the remark this answers is not in the document]
Comment (ulyssessouza) [reply pkg/compose/build_classic.go:243]: Fixed

File: pkg/compose/build_classic.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+		buildCtx = dockerfileCtx
+	}
+
+	var body io.Reader
+	if buildCtx != nil {
+		body = progress.NewProgressReader(buildCtx, progressOutput, 0, "", "Sending build context to Docker daemon")
+	}
+
+	configFile := dockerCli.ConfigFile()
+	creds, _ := configFile.GetAllCredentials()
```
[the remark this answers is not in the document]
Comment (ulyssessouza) [reply pkg/compose/build_classic.go:159]: Added a check for that one

File: pkg/compose/build_classic.go [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+			return "", errors.Errorf("error checking context: '%s'.", err)
+		}
+
+		// And canonicalize dockerfile name to a platform-independent one
+		relDockerfile = archive.CanonicalTarNameForPath(relDockerfile)
+
+		excludes = build.TrimBuildFilesFromExcludes(excludes, relDockerfile, false)
+		buildCtx, err = archive.TarWithOptions(contextDir, &archive.TarOptions{
+			ExcludePatterns: excludes,
+			ChownOpts:       &idtools.Identity{UID: 0, GID: 0},
```
[the remark this answers is not in the document]
Comment (ulyssessouza) [reply pkg/compose/build_classic.go:126]: Done

File: pkg/compose/build_classic.go [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+	// read from a directory into tar archive
+	if buildCtx == nil {
+		excludes, err := build.ReadDockerignore(contextDir)
+		if err != nil {
+			return "", err
+		}
+
+		if err := build.ValidateContextDirectory(contextDir, excludes); err != nil {
+			return "", errors.Errorf("error checking context: '%s'.", err)
```
[the remark this answers is not in the document]
Comment (ulyssessouza) [reply pkg/compose/build_classic.go:117]: Done

File: pkg/compose/build_classic.go [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+		}
+		nameDigests[name] = digest
+	}
+
+	return nameDigests, errs
+}
+
+// nolint: gocyclo
+func doBuildClassicSimpleImage(ctx context.Context, dockerCli *command.DockerCli, options buildx.Options) (string, error) {
+	var (
```
[the remark this answers is not in the document]
Comment (ulyssessouza) [reply pkg/compose/build_classic.go:60]: Yep. That's just moved code. As I was telling @ndeloof, this whole file is meant to be removed once `buildkit` supports windows containers.
So I don't think it's worth changing it's structure or anything like that.
Event the opposite, if something is fixed in `docker/cli` that would be "easier" to fix here if the functions still kind of "look like" the original ones.

File: pkg/compose/build_classic.go [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+		// Avoid leaving a stale file if we eventually fail
+		if err := os.Remove(options.ImageIDFile); err != nil && !os.IsNotExist(err) {
+			return "", errors.Wrap(err, "Removing image ID file")
+		}
+	}
+
+	switch {
+	case isLocalDir(specifiedContext):
+		contextDir, relDockerfile, err = build.GetContextFromLocalDir(specifiedContext, dockerfileName)
+		if err == nil && strings.HasPrefix(relDockerfile, ".."+string(filepath.Separator)) {
```
[the remark this answers is not in the document]
Comment (ulyssessouza) [reply pkg/compose/build_classic.go:84]: Not sure about this one. Anyways, that's the behaviour of `docker/cli`

**Comment by ulyssessouza** [comment]
@ndeloof PTAL

**Review by ndeloof — APPROVED** [review]

**Pull request merged**